File: boj/b_1931.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
arr = sorted([list(map(int, input().split())) for _ in range(n)], key=lambda x: (x[1], x[0]))
now, result = arr[0], 1
for i in range(1, n):
    if arr[i][0] >= now[1]:
        now, result = arr[i], result + 1
print(result)

File: boj/b_1931.ts
function sol1931(n: number, arr: Array<Array<number>>) {
  arr.sort((a, b) => a[1] - b[1] || a[0] - b[0]);
  let [now, result] = [arr[0], 1];
  for (let i = 1; i < n; i += 1) {
    if (arr[i][0] >= now[1]) [now, result] = [arr[i], result + 1];
  }
  console.log(result);
}
sol1931(11, [
  [1, 4],
  [3, 5],
  [0, 6],
  [5, 7],
  [3, 8],
  [5, 9],
  [6, 10],
  [8, 11],
  [8, 12],
  [2, 13],
  [12, 14],
]);
